src/units/unit.cpp

```diff
@@ -1681,9 +1681,10 @@ int unit::resistance_against(const std::string& damage_name,bool attacker,const
 {
 	int res = movement_type_.resistance_against(damage_name);
 
-	unit_ability_list resistance_abilities = get_abilities("resistance",loc, weapon, opp_weapon);
+	unit_ability_list resistance_abilities = get_abilities("resistance",loc);
 	for(unit_ability_list::iterator i = resistance_abilities.begin(); i != resistance_abilities.end();) {
-		if(!resistance_filter_matches(*i->first, attacker, damage_name, 100-res)) {
```

Comment:
i'd be better if the calling code wouldn't have to worry about doing this. mabe make two function of get_abilitites one that also does filter for weapons and one that doesn't

Reply:
@gfgtdf , but ability_affects_weapon is necessary for filtering weapon in resistance abilities, now what leadership called by function using list_ability weapon can be filtered like in abilities used like specials.
duplicate get_abilities in two version just for resistance is not a good idea for me, because this function is deprecated in other abilities.